modules/apps/27-interchain-accounts/simulation/proposals_test.go

```diff
@@ -25,32 +27,59 @@ func TestProposalMsgs(t *testing.T) {
 	ctx := sdk.NewContext(nil, cmtproto.Header{}, true, nil)
 	accounts := simtypes.RandomAccounts(r, 3)
 
-	// execute ProposalMsgs function
-	weightedProposalMsgs := simulation.ProposalMsgs()
-	require.Equal(t, 2, len(weightedProposalMsgs))
-	w0 := weightedProposalMsgs[0]
+	tests := []struct {
+		controller   *controllerkeeper.Keeper
+		host         *hostkeeper.Keeper
+		proposalMsgs int
+		isHost       []bool
```

Comment:
happy with this, but could also maybe do `expMsgs []interface` where test case 1 has `[]interface{msgHostUpdateParams, msgControllerUpdateParams}` (we can create the expected host and controller msgs at the top of the test since they are predictable)

Reply:
That's a great idea. I also updated the tests to match your others so they include a name and run in a goroutine. Let me know if there is anything else.

Reply:
Could we use `[]sdk.Msg` or `[]proto.Message` instead of `[]interface` to have stronger typing?